pad_server/src/controller.c

```diff
@@ -136,31 +139,28 @@ void *controller_run(void *arg) {
 
         /* Receive messages */
         for (;;) {
-            /* Get the message header to determine what to handle */
             header_p hdr;
-            ssize_t bread = 0;
-            size_t total_read = 0;
-
-            while (total_read < sizeof(hdr)) {
-                bread = controller_recv(&controller, (char *)&hdr + total_read, sizeof(hdr) - total_read);
-                if (bread == -1) {
-                    fprintf(stderr, "Error reading message header: %s\n", strerror(errno));
-                    break;
-                } else if (bread == 0) {
-                    fprintf(stderr, "Control box disconnected.\n");
-                    break;
-                }
-                total_read += bread;
-            }
-
-            // Error happened, do a cleanup and re-initialize connection
-            if (bread <= 0) {
+            uint8_t buffer[MAX_BUFFER_SIZE];
+            struct iovec iov[2];
```

Comment:
I think I narrowed down the problem to 2 different solutions
1. Sending the length of the payload with the header, which is not how the spec is right now
2. A recv flush function, that in case of incorrect header or error when reading header clears out the incoming buffer

Reply:
I like option 2 right now!

Reply:
> I like option 2 right now!

On it